examples/async2-passthru.rs

```diff
@@ -0,0 +1,447 @@
+use clap::Parser;
+use etherparse::{InternetSlice::*, LinkSlice::*, TransportSlice::*, *};
+use futures::{task::AtomicWaker, Future};
+use ndisapi::{EthRequest, FilterFlags};
+use std::{
+    ffi::c_void,
+    pin::Pin,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    task::{Context, Poll},
+};
+use tokio::{runtime::Builder, sync::oneshot};
+use windows::{
+    core::Result,
+    Win32::{
+        Foundation::{GetLastError, CloseHandle, BOOLEAN, HANDLE},
+        System::Threading::{
+            CreateEventW, RegisterWaitForSingleObject, ResetEvent, UnregisterWaitEx, INFINITE,
+            WT_EXECUTEINWAITTHREAD,
+        },
+    },
+};
+
+// The struct NdisapiAdapter represents a network adapter with its associated driver and relevant handles.
+pub struct NdisapiAdapter {
+    driver: Arc<ndisapi::Ndisapi>, // The network driver for the adapter.
+    adapter_handle: HANDLE,        // The handle of the network adapter.
+    notif: Win32EventFuture,       // A future that resolves when a Win32 event is signaled.
```

Comment:
Nit: These aren't rustdoc, consider the following:

```suggestion
    /// The network driver for the adapter.
    driver: Arc<ndisapi::Ndisapi>, 
    /// The handle of the network adapter.
    adapter_handle: HANDLE,
    /// A future that resolves when a Win32 event is signaled.
    notif: Win32EventFuture,
```